pkg/adapter/metrics.go

```diff
@@ -69,7 +69,19 @@ type (
 		Builder
 
 		// NewMetrics returns a new instance of the Metrics aspect.
-		NewMetrics(env Env, config AspectConfig) (MetricsAspect, error)
+		NewMetrics(env Env, config AspectConfig, metrics []MetricDefinition) (MetricsAspect, error)
+	}
+
+	// MetricDefinition provides the basic description of a metric schema
+	// for which metrics adapters will be sent Values at runtime.
+	MetricDefinition struct {
+		// Name is the canonical name of the metric.
+		Name string
+		// Kind provides type information about the metric.
+		Kind Kind
+		// Labels are the names of keys for dimensional data that will
+		// generated at runtime and passed along with metric values.
+		Labels []string
```

Comment:
@ZackButcher labels are defined in the metric descriptor. they are a different word for attribute name/value pairs. so, the manager would read the descriptor, get a list of attr names, look them up in the attr.Bag and put them in the Labels map and pass them along.